core/src/replay_stage.rs

```diff
@@ -297,6 +297,11 @@ pub struct ReplayStageConfig {
     pub replay_slots_concurrently: bool,
 }
 
+struct ReplayStageState {
+    pub blockstore: Arc<Blockstore>,
+    pub bank_forks: Arc<RwLock<BankForks>>,
+}
+
```

Comment:
@AshwinSekar and @carllin - This will be tedious work so I wanted to get some opinions and get agreement on general path forward before touching just about every function in `ReplayStage`.

General idea would be to pull most of the stuff from `ReplayStageConfig` + everything else we create before we enter the main loop into this struct, and then access it from all the sub-functions as if it were a `self`.

Thoughts ? I think putting this stuff as members of `ReplayStage` could be problematic in that we wouldn't drop things until we call `.join()`. So, seemingly better to hold a separate struct but I'll think on this a bit more as well